File: leet_0916_string_subset___int26.java
class Solution {
    public List<String> wordSubsets(String[] words1, String[] words2)
    {
        // String[]    res; /* you need List<String> */
        List<String>    res;
        int             freq[];
        int             len1, len2, i;

        res = new ArrayList<String>();
        len1 = words1.length;
        len2 = words2.length;
        freq = new int [26];
        Arrays.fill(freq, 0);
        for (String s: words2)
        {
            int     temp[] = new int [26];

            Arrays.fill(temp, 0);
            for (char c: s)
            {
                temp[c - 'a']++;
                freq[c - 'a'] = max(freq[c - 'a'], temp[c - 'a']);
            }
        }
        for (String s: words1)
        {
            int     temp[] = new int [26];

            Arrays.fill(temp, 0);
            for (char c: s)
                temp[c - 'a']++;
            i = -1;
            while (++i < 26 && temp[i] >= freq[i])
                ;
            if (i == 26)
                res.add(s);
        }
    }
}

File: leet_0916_string_subset___int26.js
/**
 * @param {string[]} words1
 * @param {string[]} words2
 * @return {string[]}
 */
var wordSubsets = function(words1, words2) {
    res = []
    n1 = words1.length
    n2 = words2.length
    a = 'a'.charCodeAt(0)
    freq = new Array(26).fill(0)
    for (let s of words2) {
        temp = new Array(26).fill(0)
        for (let c of s) {
            asc = c.charCodeAt(0) - a
            temp[asc] ++
            freq[asc] = Math.max(freq[asc], temp[asc])
        }
    }
    for (let s of words1) {
        temp = new Array(26).fill(0)
        for (let c of s) {
            temp[c.charCodeAt(0) - a] ++
        }
        i = -1
        while (++i < 26 && temp[i] >= freq[i])
            ;
        if (i == 26)
            res.push(s)
    }
    return res
};
